tests/manifests/test_configs_consistency.py

```diff
@@ -0,0 +1,524 @@
+# Copyright 2024-2025 New Vector Ltd
+#
+# SPDX-License-Identifier: AGPL-3.0-only
+
+import abc
+import re
+from base64 import b64decode
+from dataclasses import dataclass, field
+
+import pytest
+
+from . import DeployableDetails, secret_values_files_to_test, values_files_to_test
+from .test_configs_and_mounts_consistency import (
+    assert_exists_according_to_hook_weight,
+    find_keys_mounts_in_content,
+    get_configmap,
+    get_or_empty,
+    get_secret,
+    get_volume_from_mount,
+    match_path_in_content,
+)
+from .utils import template_id, template_to_deployable_details
+
+
+## Gets all mounted files in a given container from the ConfigMaps referenced by the volume mounts
+def get_all_mounted_files(workload_spec, container_name, templates):
+    found_files = {}
+    for container_spec in workload_spec["containers"] + workload_spec.get("initContainers", {}):
```

Comment:
```suggestion
    for workload_spec.get("initContainers", []) + container_spec in workload_spec["containers"]:
```

As `initContainers` are a list and come first. This function should probably also short-circuit once it has processed the desired container